com.unity.render-pipelines.core/Runtime/Lighting/ProbeVolume/ProbeBrickPool.cs

```diff
@@ -264,59 +261,61 @@ internal static Vector3Int ProbeCountToDataLocSize(int numProbes)
             return new Vector3Int(width, height, depth);
         }
 
-        public static DataLocation CreateDataLocation(int numProbes, bool compressed, ProbeVolumeSHBands bands, string name, out int allocatedBytes)
+        public static Texture CreateDataTexture(int width, int height, int depth, GraphicsFormat format, string name, bool allocateRendertexture, ref int allocatedBytes)
+        {
+            int elementSize = format == GraphicsFormat.R16G16B16A16_SFloat ? 8 :
+                format == GraphicsFormat.R8G8B8A8_UNorm ? 4 : 1;
+
+            Texture texture;
+            allocatedBytes += (width * height * depth) * elementSize;
+            if (allocateRendertexture)
+            {
+                texture = new RenderTexture(new RenderTextureDescriptor()
+                {
+                    width = width,
+                    height = height,
+                    volumeDepth = depth,
+                    graphicsFormat = format,
+                    mipCount = 1,
+                    enableRandomWrite = true,
+                    dimension = TextureDimension.Tex3D,
+                    msaaSamples = 1,
+                });
+            }
+            else
+                texture = new Texture3D(width, height, depth, format, TextureCreationFlags.None, 1);
+
+            texture.hideFlags = HideFlags.HideAndDontSave;
+            texture.name = name;
+
+            if (allocateRendertexture)
+                (texture as RenderTexture).Create();
+            return texture;
+        }
+
+        public static DataLocation CreateDataLocation(int numProbes, bool compressed, ProbeVolumeSHBands bands, string name, bool allocateRendertexture, out int allocatedBytes)
         {
             Vector3Int locSize = ProbeCountToDataLocSize(numProbes);
             int width = locSize.x;
             int height = locSize.y;
             int depth = locSize.z;
 
-            int texelCount = width * height * depth;
-
             DataLocation loc;
+            var primaryFormat = GraphicsFormat.R16G16B16A16_SFloat;
```

Comment:
why primary format? I would put L0 texture format here, nothing primary about it